server/src/main/java/org/apache/druid/server/coordinator/duty/KillUnusedSegments.java

```diff
@@ -118,17 +129,48 @@ public DruidCoordinatorRuntimeParams run(DruidCoordinatorRuntimeParams params)
     return params;
   }
 
+  /**
+   * For a given datasource and limit of segments that can be killed in one task, determine the interval to be
+   * submitted with the kill task.
+   *
+   * @param dataSource dataSource whose unused segments are being killed.
+   * @param limit the maximum number of segments that can be included in the kill task.
+   * @return {@link Interval} to be used in the kill task.
+   */
   @VisibleForTesting
   @Nullable
   Interval findIntervalForKill(String dataSource, int limit)
   {
     List<Interval> unusedSegmentIntervals =
-        segmentsMetadataManager.getUnusedSegmentIntervals(dataSource, DateTimes.nowUtc().minus(retainDuration), limit);
+        segmentsMetadataManager.getUnusedSegmentIntervals(dataSource, getEndTimeUpperLimit(), limit);
 
     if (unusedSegmentIntervals != null && unusedSegmentIntervals.size() > 0) {
       return JodaUtils.umbrellaInterval(unusedSegmentIntervals);
     } else {
       return null;
     }
   }
+
+  /**
+   * Calculate the {@link DateTime} that wil form the upper bound when looking for segments that are
+   * eligible to be killed. If ignoreDurationToRetain is true, we have no upper bound and return a DateTime object
+   * for 9999-12-31T23:59. This static date has to be used because the metastore is not comparing date objects, but rather
+   * varchar columns. This means DateTimes.MAX is less than the 21st century and beyond for according to the metastore, due to its
+   * year starting with a '1'
+   *
+   * @return {@link DateTime} representing the upper bound time used when looking for segments to kill.
+   */
+  @VisibleForTesting
+  DateTime getEndTimeUpperLimit()
+  {
+    return ignoreRetainDuration
+           ? DateTimes.of(9999, 12, 31, 23, 59)
```

Comment:
Can we use `DateTimes.CAN_COMPARE_AS_YEAR_MAX` instead? Might be worth renaming that property as well for readability purposes.